components/scripts/gradle/03-validate-local-build-caching-different-locations.sh

```diff
@@ -158,8 +170,33 @@ execute_second_build() {
 }
 
 fetch_build_cache_metrics() {
-  read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  if [ "$build_scan_publishing_mode" == "on" ]; then
```

Comment:
0: When it comes time to add support to other experiments, we could move this logic into either a new shared function, or we could update the existing read and fetch functions to behave differently if build scan publishing is disabled.